docs/tutorials/gluon/fit_api_tutorial.md

```diff
@@ -0,0 +1,311 @@
+train_first_elem, train_first_label = fashion_mnist_train[0]
+print ("Shape of each iamge : ", train_first_elem.shape)
+```
+
+    Number of training samples : 60000 <!--notebook-skip-line-->
+    Number of validation samples : 10000 <!--notebook-skip-line-->
+    Shape of each iamge :  (28, 28, 1) <!--notebook-skip-line-->
+
+
+Now let's try to visualize the dataset before we proceed further
+
+
+```python
+from IPython import display
+import matplotlib.pyplot as plt
+
+# Function to display the data
+def show_fashion_mnist_data(images, labels):
+    display.set_matplotlib_formats('svg')
+    _, figs = plt.subplots(1, len(images), figsize=(12, 12))
+    
+    for figure, x, y in zip(figs, images, labels):
+        figure.imshow(x.reshape((28, 28)).asnumpy())
+        axes = figure.axes
+        axes.set_title(text_labels[int(y)])
+        axes.title.set_fontsize(12)
+        axes.get_xaxis().set_visible(False)
+        axes.get_yaxis().set_visible(False)
+    
+    plt.show()
+```
+
+
+```python
+images, labels = fashion_mnist_train[0:10]
+show_fashion_mnist_data(images, labels)
+```
+
+
+![png](https://raw.githubusercontent.com/piyushghai/web-data/master/mxnet/doc/tutorials/gluon/fashion_mnist.png)<!--notebook-skip-line-->
+
+
+## Pre-processing the data
+
+In order to prepare our data to training the model, we will perform a few pre-processing steps on the dataset. We will :
+- Resize the image
+- Convert the pixel values in the image from (0 to 255) to (0 to 1)
+- Normalize the images with mean 0 and variance 1. 
+
+We will be using ```gluon.data.vision.tranforms``` which provides out of the box transformation APIs.
+To read more about the available transformations, check out [the official documentation](https://mxnet.incubator.apache.org/api/python/gluon/data.html#vision-transforms).
+
+
+```python
+transformers = [gluon.data.vision.transforms.Resize(224), # We pick 224 as the model we use takes an input of size 224.
+                gluon.data.vision.transforms.ToTensor(), 
+                gluon.data.vision.transforms.Normalize(mean = 0, std = 1)]
+
+# Now we will stack all these together.
+transform = gluon.data.vision.transforms.Compose(transformers)
+```
+
+
+```python
+# Apply the transformations
+fashion_mnist_train = fashion_mnist_train.transform_first(transform)
+fashion_mnist_val = fashion_mnist_val.transform_first(transform)
+```
+
+## Data Loaders
+
+In order to feed the data to our model, we need to use a [Data Loader](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=dataloader#mxnet.gluon.data.DataLoader). 
+
+Data Loaders are used to read the dataset, create mini-batches from the dataset and feed them to the neural network for training.
+
+
+```python
+batch_size = 256 # Batch size of the images
+num_workers = 4 # The number of parallel workers for loading the data using Data Loaders.
+
+train_iter = gluon.data.DataLoader(fashion_mnist_train, batch_size=batch_size, shuffle=True, num_workers=num_workers)
```

Comment:
MXNet also has DataIter class(mx.io.DataIter). It would be good if you change the naming as 'train_data_loader' and 'train_val_loader' to avoid confusion.

Reply:
Done in e44ddaf33e207359217061b2b4c17d723a25f6f0